KB: Multiple child Grids per item (#745)
telerik/blazor-docs: knowledge-base/grid-multiple-child-grids-per-item.md

---

> title: Multiple child Grids per item

Reviewer: ```suggestion
title: Multiple Details Grids per Master
```

> description: How to render more than one child Grid for an item?

Reviewer: ```suggestion
description: How to render multiple child (detail) Grids under a master Grid?
```

type: how-to

> page_title: Multiple child Grids per item

Reviewer: ```suggestion
page_title: Multiple Detail Grids per Master
```

slug: grid-kb-multiple-child-grids-per-item
position: 
tags: grid, hierarchy, multiple, children
ticketid: 1550340
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Grid for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

> We have a hierarchical Grid where each item has two sub-Grids. We want to display both child Grids simultaneously.

Reviewer: ```suggestion
We have a hierarchical Grid where each master item has two child sub-Grids. We want to display both child Grids at the same time.
```

## Solution

> The [hierarchy in Grid]({%slug components/grid/features/hierarchy%}) is achieved through declaring a child Grid in a [`DetailTemplate`](https://docs.telerik.com/blazor-ui/api/Telerik.Blazor.Components.TelerikGrid-1#collapsible-Telerik_Blazor_Components_TelerikGrid_1_DetailTemplate). As the `DetailTemplate` is essentially a `RenderFragment`, you can, in fact, render any desired content there - be that one, two or more Grids, other components or html elements.

Reviewer: ```suggestion
[Grid hierarchy]({%slug components/grid/features/hierarchy%}) requires a child Grid in a [`DetailTemplate`](https://docs.telerik.com/blazor-ui/api/Telerik.Blazor.Components.TelerikGrid-1#collapsible-Telerik_Blazor_Components_TelerikGrid_1_DetailTemplate). The `DetailTemplate` is essentially a `RenderFragment`, so it can include any desired content there - be that one, two or more Grids, other components or html elements.
```

> In order to render two child Grids at the same time, you just need to declare both of them under the `DetailTemplate` tag. The example below demonstrates how to achieve this.

Reviewer: ```suggestion
In order to render two child (detail) Grids at the same time, you just need to declare both of them inside the `DetailTemplate` tag. The example below demonstrates how to achieve this.
```
